Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2027,8 +2029,23 @@ String generateShortLinkInternal(ServerRequestCreateUrl req) {
         }
         return null;
     }
-    
-    
+
+    /**
+     * <p>Gets the Branch Internal App Configuration for the last given live key</p>
+     *
+     */
+
+    JSONObject generateAppConfigInternal() {
+        ServerResponse response = null;
+        try {
+            response = new getAppConfigTask().execute().get(PrefHelper.TIMEOUT, TimeUnit.MILLISECONDS);
```

Comment:
This Involves a blocking call. `getAppConfigTask.excute.get()` Waits if necessary for at most the given time for the computation. PrefHelper.TIMEOUT is `5500` sec. This can be potential ANR candidate (Since ANR watch dog timer is 5 sec)

Reply:
Synced offline. Since this call is not meant to be made on the production deployment we should be okay with this.